config/configmapprovider/expand_test.go

```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/stretchr/testify/require"
 
 	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/filemapprovider"
```

Comment:
Any reasons to not have it as `config/mapprovider/file`?

Reply:
No reason... Then New should be NewMapProvider? Since will be used as file.NewMapProvider

Reply:
So we have few options:
1. `config/mapprovider/file`, `config/mapprovider/env`, etc.
2. `configmapprovider/file`, `configmapprovider/env`, etc.
3. `configmapprovider/filemapprovider`, `configmapprovider/envmapprovider`, etc. - This is kind of consistent with the other components, `exporter/otlpexporter`.

I kind of prefer 3 actually.

Reply:
Based on this, I see another option:
4. `config/mapprovider/filemapprovider`, `config/mapprovider/envmapprovider`, etc.

Reply:
So there are 2 decisions:
1. `config/mapprovider/` vs `configmapprovider/`?
2. `file` vs `filemapprovider` as the package?

Reply:
My vote is for `config/mapprovider/filemapprovide` to avoid the "import" rename, since for example "file" package is used a lot.

/cc @open-telemetry/collector-approvers @open-telemetry/collector-maintainers please let me know your preference.

Reply:
I think that `config/mapprovider/filemapprovider` would be my preference, though I could live with `config/mapprovider/file` as well.

Reply:
We have enough votes for `filemapprovider` so going with that.